src/platform/MpiServer.c

```diff
@@ -0,0 +1,611 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <pthread.h>
+
+#include <PlatformCommon.h>
+#include <MpiServer.h>
+
+static const char* g_socketPrefix = "/run/osconfig";
+static const char* g_mpiSocket = "/run/osconfig/mpid.sock";
+
+static const char* g_clientName = "ClientName";
+static const char* g_maxPayloadSizeBytes = "MaxPayloadSizeBytes";
+static const char* g_clientSession = "ClientSession";
+static const char* g_componentName = "ComponentName";
+static const char* g_objectName = "ObjectName";
+static const char* g_payload = "Payload";
+
+#define MAX_CONTENTLENGTH_LENGTH 16
+#define MAX_REASON_PHRASE_LENGTH 32
+#define MAX_STATUS_CODE_LENGTH 3
+
+static int g_socketfd;
+static struct sockaddr_un g_socketaddr;
+static socklen_t g_socketlen;
+
+static pthread_t g_worker;
+static bool g_serverActive = false;
+
+typedef enum HTTP_STATUS
+{
+    HTTP_OK = 200,
+    HTTP_BAD_REQUEST = 400,
+    HTTP_NOT_FOUND = 404,
+    HTTP_INTERNAL_SERVER_ERROR = 500
+} HTTP_STATUS;
+
+static void GetHttpReasonPhrase(HTTP_STATUS statusCode, char* phrase)
```

Comment:
>GetHttpReasonPhrase [](http://example.com/codeflow?start=12&length=19)

That is not a phrase. Also the Get seems out of place. Suggest: "HttpReasonAsString"